lib/fabrics/ofi/src/internal/PassiveEndpoint.hpp

```diff
@@ -0,0 +1,116 @@
+// SPDX-FileCopyrightText: 2025 Contributors to the Media eXchange Layer project.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <rdma/fi_endpoint.h>
+#include "Address.hpp"
+#include "Endpoint.hpp"
+#include "Event.hpp"
+#include "EventQueue.hpp"
+#include "Fabric.hpp"
+
+namespace mxl::lib::fabrics::ofi
+{
+    /** \brief RAII Wrapper around a libfabric passive endpoint (`fid_pep`).
+     *
+     * A passive endpoint can be viewed like a bound TCP socket listening for incoming connections.
+     */
+    class PassiveEndpoint
+    {
+    public:
+        ~PassiveEndpoint();
+
+        // Copy constructor and assignment operators are deleted
+        PassiveEndpoint(PassiveEndpoint const&) = delete;
+        void operator=(PassiveEndpoint const&) = delete;
+
+        // Implement move constructor and assignment operator
+        PassiveEndpoint(PassiveEndpoint&&) noexcept;
+        PassiveEndpoint& operator=(PassiveEndpoint&&);
+
+        /** \brief Allocate a new PassiveEndpoint associated with the given Fabric.
+         *
+         * \param fabric The Fabric to create the PassiveEndpoint on.
+         * \return A new PassiveEndpoint object.
+         */
+        static PassiveEndpoint create(std::shared_ptr<Fabric>);
```

Comment:
Should be annotated with `[[nodiscard]]`. Also method/function signatures in headers should name their arguments.